apps/spruce/src/pages/waterfall/BuildRow.tsx

```diff
@@ -0,0 +1,168 @@
+import styled from "@emotion/styled";
+import { palette } from "@leafygreen-ui/palette";
+import { Link } from "react-router-dom";
+import { TaskStatus } from "@evg-ui/lib/types/task";
+import { useWaterfallAnalytics } from "analytics";
+import { StyledLink } from "components/styles";
+import { getTaskRoute, getVariantHistoryRoute } from "constants/routes";
+import { size } from "constants/tokens";
+import {
+  WaterfallBuild,
+  WaterfallBuildVariant,
+  WaterfallQuery,
+} from "gql/generated/types";
+import {
+  BuildVariantTitle,
+  columnBasis,
+  gridGroupCss,
+  InactiveVersion,
+  Row,
+} from "./styles";
+
+const { black, gray, green, white } = palette;
+
+export const BuildRow: React.FC<{
+  build: WaterfallBuildVariant;
+  projectIdentifier: string;
+  versions: WaterfallQuery["waterfall"]["versions"];
+}> = ({ build, projectIdentifier, versions }) => {
+  const { sendEvent } = useWaterfallAnalytics();
+  const handleVariantClick = useCallback(
+    () => sendEvent({ name: "Clicked variant label" }),
+    [sendEvent],
+  );
+  const handleTaskClick = useCallback(
+    (status: string) => () =>
+      sendEvent({
+        name: "Clicked task box",
+        "task.status": status,
+      }),
+    [sendEvent],
+  );
+
+  const { builds, displayName } = build;
+  let buildIndex = 0;
+  return (
+    <Row>
+      <BuildVariantTitle>
+        <StyledLink
+          href={getVariantHistoryRoute(projectIdentifier, build.id)}
+          onClick={handleVariantClick}
+        >
+          {displayName}
+        </StyledLink>
+      </BuildVariantTitle>
+      <BuildGroup data-cy="build-group">
+        {versions.map(({ inactiveVersions, version }) => {
+          if (inactiveVersions?.length) {
+            return (
+              <InactiveVersion
+                key={inactiveVersions[0].id}
+                data-cy="inactive-column"
+              />
+            );
+          }
+          /* The list of builds returned does not include a placeholder for inactive builds, so we need to check whether the build matches the version in the current column.
+        Builds are sorted in descending revision order and so match the versions' sort order. */
+          if (version && version.id === builds?.[buildIndex]?.version) {
+            const b = builds[buildIndex];
+            buildIndex += 1;
+            return (
+              <BuildGrid
+                key={b.id}
+                build={b}
+                handleTaskClick={handleTaskClick}
+              />
+            );
+          }
+          return <Build key={version?.id} />;
+        })}
+      </BuildGroup>
+    </Row>
+  );
+};
+
+const BuildGrid: React.FC<{
+  build: WaterfallBuild;
+  handleTaskClick: (s: string) => () => void;
+}> = ({ build, handleTaskClick }) => (
+  <Build>
+    {build.tasks.map(({ displayName, id, status }) => (
+      <SquareMemo
+        key={id}
+        data-tooltip={displayName}
+        onClick={handleTaskClick(status)}
```

Comment:
You can decrease the number of event handlers by adding the onClick to `Build`. The callback would look like this:
```
(event) => {
    handleTaskClick(event.target?.getAttribute('status'))
}
```
I can't really notice a difference in render speed but this technically would reduce memory usage.